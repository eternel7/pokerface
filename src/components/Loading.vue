<template>
  <div v-if="visible" id="full_page_overlay">
    <div id="loading_in_progress" class="mdl-color--accent"></div>
  </div>
</template>

<script>
  export default {
    name: 'Loading',
    computed: {
      visible: function () {
        return !!this.$root.loading
      }
    }
  }
</script>

<style scoped>
  #loading_in_progress {
    position: absolute;
    left: 50%;
    top: 50%;
    margin: -20px 0 0 -20px;
    height: 40px;
    width: 40px;
    border-radius: 50%;
    opacity: 1;

    animation: pulsate 1s ease-out;
    animation-iteration-count: infinite;
  }

  @keyframes pulsate {
    0% {
      transform: perspective(120px) rotateX(0deg) rotateY(0deg);
    }
    50% {
      transform: perspective(120px) rotateX(-180deg) rotateY(0deg);
    }
    100% {
      transform: perspective(120px) rotateX(-180deg) rotateY(-180deg);
    }
  }

  #full_page_overlay {
    background: rgba(0,0,0, 0.5);
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    z-index: 9999;
  }
</style>
