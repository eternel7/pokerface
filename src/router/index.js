import Vue from 'vue'
import Router from 'vue-router'
import { authMixin } from '@/auth/authMixin.js'
import Home from '@/components/pages/Home'
import SignInPage from '@/components/pages/Sign-in'
import SignUp from '@/components/pages/Sign-up'
import Profile from '@/components/pages/Profile'
import ForgotPassword from '@/components/pages/Forgot-password'
import ResetPassword from '@/components/pages/Reset-password'
import CookiesPolicy from '@/components/pages/Cookies-Policy'
import PrivacyStatement from '@/components/pages/Privacy-Statement'
import Chat from '@/components/pages/Chat'
import Chatrooms from '@/components/pages/Chatrooms'

Vue.use(Router)

export default new Router({
  routes: [
    {
      path: '/',
      name: 'Home',
      component: Home
    },
    {
      path: '/signin',
      name: 'Sign in',
      component: SignInPage
    },
    {
      path: '/signup',
      name: 'Sign up',
      component: SignUp
    },
    {
      path: '/profile',
      name: 'Profile',
      component: Profile,
      beforeEnter: (to, from, next) => {
        authMixin.methods.checkToken('all', next)
      }
    },
    {
      path: '/forgotpassword',
      name: 'Forgotpassword',
      component: ForgotPassword
    },
    {
      path: '/resetpassword/:resetPasswordToken?',
      name: 'ResetPassword',
      component: ResetPassword
    },
    {
      path: '/cookiespolicy',
      name: 'Cookies policy',
      component: CookiesPolicy
    },
    {
      path: '/privacystatement',
      name: 'Privacy statement',
      component: PrivacyStatement
    },
    {
      path: '/chat/:id',
      name: 'Chat',
      component: Chat,
      beforeEnter: (to, from, next) => {
        authMixin.methods.checkToken('all', next)
      }
    },
    {
      path: '/chatrooms',
      name: 'Chatrooms',
      component: Chatrooms,
      beforeEnter: (to, from, next) => {
        authMixin.methods.checkToken('all', next)
      }
    }
  ]
})
