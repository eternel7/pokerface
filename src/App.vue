<template>
  <div id="app-content" class="mdl-layout mdl-js-layout mdl-layout--fixed-header">
    <loading></loading>
    <transition name="fade-slide-up" mode="out-in">
      <header v-show="!!headerAvailable" class="mdl-layout__header">
        <div aria-expanded="false" role="button" tabindex="10"
             class="mdl-layout__drawer-button" v-on:click="back">
          <transition name="fade" mode="out-in">
            <i v-if="backAvailable" key="arrow_back" class="material-icons">arrow_back</i>
            <i v-else key="menu" class="material-icons">menu</i>
          </transition>
        </div>
        <div class="mdl-layout__header-row">
          <span class="mdl-layout-title">Pokerface</span>
          <div class="mdl-layout-spacer"></div>
          <transition name="fade">
            <div v-show="searchAvailable" class="mdl-textfield mdl-js-textfield mdl-textfield--expandable
                  mdl-textfield--floating-label mdl-textfield--align-right">
              <label class="mdl-button mdl-js-button mdl-button--icon"
                     for="fixed-header-drawer-exp">
                <i class="material-icons">search</i>
              </label>
              <div class="mdl-textfield__expandable-holder">
                <input class="mdl-textfield__input" type="text" name="search"
                       id="fixed-header-drawer-exp">
              </div>
            </div>
          </transition>
        </div>
      </header>
    </transition>
    <navbar v-bind:backAvailable="backAvailable"></navbar>
    <main class="mdl-layout__content">
      <section class="section">
        <div class="section-inner">
          <transition name="fade-slide-right" mode="out-in">
            <router-view></router-view>
          </transition>
        </div>
      </section>
      <div id="toast" class="mdl-js-snackbar mdl-snackbar">
        <div class="mdl-snackbar__text"></div>
        <button class="mdl-snackbar__action" type="button"></button>
      </div>
    </main>
  </div>
</template>

<script>
  import PageBase from '@/components/pages/Page'
  import Loading from '@/components/Loading'
  import NavBar from '@/components/Navbar'

  export default {
    name: 'app',
    props: ['message', 'searchAvailable', 'backAvailable', 'headerAvailable'],
    extends: PageBase,
    components: {
      navbar: NavBar,
      loading: Loading
    },
    data () {
      return {}
    },
    methods: {
      showSnackbar: function () {
        let snackbarContainer = document.querySelector('#toast')
        snackbarContainer.MaterialSnackbar.showSnackbar(this.$root)
      },
      nothing: function (evt) {
      },
      back: function (evt) {
        if (this.headerAvailable && this.backAvailable) {
          let layout = document.querySelector('.mdl-layout')
          layout.MaterialLayout.toggleDrawer()
          history.go(-1)
        }
      }
    }
  }
</script>

<style>
  #app-content {
    background: #585858;
  }

  .fade-enter-active, .fade-leave-active {
    transition: opacity .5s;
  }

  .fade-enter, .fade-leave-to /* .fade-leave-active below version 2.1.8 */
  {
    opacity: 0;
  }

  .fade-slide-right-enter-active {
    transition: all 0.5s ease;
  }

  .fade-slide-right-leave-active {
    transition: all 0.5s ease;
  }

  .fade-slide-right-enter, .fade-slide-right-leave-to {
    transform: translateX(40px);
    opacity: 0;
  }

  .fade-slide-up-enter-active {
    transition: all 0.5s ease;
  }

  .fade-slide-up-leave-active {
    transition: all 0.5s ease;
  }

  .fade-slide-up-enter, .fade-slide-up-leave-to {
    transform: translateY(-40px) scaleY(0);
    opacity: 0;
  }
</style>
