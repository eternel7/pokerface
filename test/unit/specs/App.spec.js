import {shallowMount} from '@vue/test-utils'
import App from '@/App'
import router from '@/router'
import Loading from '@/components/Loading'
import NavBar from '@/components/Navbar'

describe('App.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(App, {router})
  })

  it('should match Snapshot', () => {
    expect(wrapper.element).toBeDefined()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should contain Loading component', () => {
    expect(wrapper.contains(Loading)).toBeTruthy()
  })

  it('should contain Navbar component', () => {
    expect(wrapper.contains(NavBar)).toBeTruthy()
  })
})
