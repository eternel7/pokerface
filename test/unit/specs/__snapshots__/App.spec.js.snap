// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue should match Snapshot 1`] = `
<div
  class="mdl-layout mdl-js-layout mdl-layout--fixed-header"
  id="app-content"
>
  <loading-stub />
   
  <header
    class="mdl-layout__header"
    mode="out-in"
    name="fade-slide-up"
    style="display: none;"
  >
    <div
      aria-expanded="false"
      class="mdl-layout__drawer-button"
      role="button"
      tabindex="10"
    >
      <i
        class="material-icons"
        mode="out-in"
        name="fade"
      >
        menu
      </i>
    </div>
     
    <div
      class="mdl-layout__header-row"
    >
      <span
        class="mdl-layout-title"
      >
        Pokerface
      </span>
       
      <div
        class="mdl-layout-spacer"
      />
       
      <div
        class="mdl-textfield mdl-js-textfield mdl-textfield--expandable
                mdl-textfield--floating-label mdl-textfield--align-right"
        name="fade"
        style="display: none;"
      >
        <label
          class="mdl-button mdl-js-button mdl-button--icon"
          for="fixed-header-drawer-exp"
        >
          <i
            class="material-icons"
          >
            search
          </i>
        </label>
         
        <div
          class="mdl-textfield__expandable-holder"
        >
          <input
            class="mdl-textfield__input"
            id="fixed-header-drawer-exp"
            name="search"
            type="text"
          />
        </div>
      </div>
    </div>
  </header>
   
  <navbar-stub />
   
  <main
    class="mdl-layout__content"
  >
    <section
      class="section"
    >
      <div
        class="section-inner"
      >
        <router-view-stub
          mode="out-in"
          name="fade-slide-right"
        />
      </div>
    </section>
     
    <div
      class="mdl-js-snackbar mdl-snackbar"
      id="toast"
    >
      <div
        class="mdl-snackbar__text"
      />
       
      <button
        class="mdl-snackbar__action"
        type="button"
      />
    </div>
  </main>
</div>
`;
